kolibri/core/assets/src/views/breadcrumbs/index.vue

```diff
@@ -0,0 +1,254 @@
+<template>
+
+  <nav class="breadcrumbs">
+
+    <div v-if="collapsedCrumbs.length" class="breadcrumbs-dropdown-wrapper">
+      <ui-icon-button :has-dropdown="true" icon="expand_more" size="small">
+        <div slot="dropdown" class="breadcrumbs-dropdown">
+          <ol class="breadcrumbs-dropdown-items">
+            <li v-for="crumb in collapsedCrumbs" class="breadcrumbs-dropdown-item">
+              <router-link :to="crumb.link" :style="{ maxWidth: `${collapsedCrumbMaxWidth}px` }">
+                {{ crumb.text }}
+              </router-link>
+            </li>
+          </ol>
+        </div>
+      </ui-icon-button>
+    </div>
+
+    <ol class="breadcrumbs-visible-items">
+      <template v-for="(crumb, index) in crumbs">
+        <li
+          v-if="index !== crumbs.length - 1"
+          :ref="`crumb${index}`"
+          class="breadcrumbs-visible-item breadcrumbs-visible-item-notlast"
+          v-show="!crumb.collapsed"
+         >
+          <router-link :to="crumb.link">{{ crumb.text }}</router-link>
+        </li>
+
+        <li
+          v-else
+          :ref="`crumb${index}`"
+          class="breadcrumbs-visible-item breadcrumb-visible-item-last"
+        >
+          <span :style="{ maxWidth: `${lastCrumbMaxWidth}px` }">{{ crumb.text }}</span>
+        </li>
+      </template>
+    </ol>
+
+  </nav>
+
+</template>
+
+
+<script>
+
+  const ResizeSensor = require('css-element-queries/src/ResizeSensor');
+  const ResponsiveElement = require('kolibri.coreVue.mixins.responsiveElement');
+  const ValidateLinkObject = require('kolibri.utils.validateLinkObject');
+  const filter = require('lodash/filter');
+  const startsWith = require('lodash/startsWith');
+  const throttle = require('lodash/throttle');
+
+  const DROPDOWN_BTN_WIDTH = 55;
+  const DROPDOWN_SIDE_PADDING = 32; // pulled from .breadcrumbs-dropdown
+  const MAX_CRUMB_WIDTH = 300; // pulled from .breadcrumbs-visible-item class
+
+  module.exports = {
+    $trNameSpace: 'breadcrumbs',
+
+    mixins: [ResponsiveElement],
+
+    components: {
+      'ui-icon-button': require('keen-ui/src/UiIconButton'),
+    },
+
+    props: {
+      items: {
+        type: Array,
+        required: true,
+        validator(crumbItems) {
+          const crumbs = Array.from(crumbItems);
+          // Must not be empty
+          if (!crumbs.length) {
+            return false;
+          }
+          // All must have text
+          if (!crumbs.every(crumb => Boolean(crumb.text))) {
+            return false;
+          }
+          crumbs.pop();
+          // All, but the last, must have a valid router link
+          return crumbs.every(crumb => ValidateLinkObject(crumb.link));
```

Comment:
Could also be written

```javascript
crumbItems.slice(0, -1).every(...)
```

and then the `Array.from` and `crumbs` temp var can be omitted.